ursula monitoring update for centos

Xiaohua-Shen opened the pull request:
related change works both for centos-newton and ubuntu-mitaka.
1. novaclient old version support parameter project_id, new version change to project_name.
solution: import shade, which works for both version. (Cloud configs are read with os-client-config)
http://docs.openstack.org/infra/shade/
it's used in other metrics plugs in too.

2. ImportError: cannot import name environment for keystone token expiration check.
new version of keystone has no environment directory anymore.
solution: remove environement, and it has no impact to both old ubuntu cloud and new centos cloud

3. check-mem.sh not work for centos, since the free -m output is different.
solution: get the latest version of sensu-plugin community version, and works

4. check-nbd-nonunique.sh not work well both for unbuntu and centos.
The purpose of this check is for qemu-nbd related process only, but existing shell didn't consider situation of no qemu-nbd related process exist.

5. metrics-disk-capacity, use the community version to correct rhel related issue.

6. metrics-os-api keystone url incorrect, 404 not found

Xiaohua-Shen committed:
ursula monitoring update for centos

related change works both for centos-newton and ubuntu-mitaka

scoop206 reviewed (approved):
looks good :+1:
I like the use of shade and how we've pulled in community updates to fix some of the centos issues

paulczar commented:
looks okay to me ... is changing the `#!` necessary ?

scoop206 commented:
for the `#!` change perhaps instead we could update the sensu user's PATH to include /opt/sensu/embedded/bin ?

Xiaohua-Shen commented:
add '/opt/sensu/embedded/bin' to the begining of PATH seens work. 
```
[root@controller-0 checks]# more /etc/default/sensu
EMBEDDED_RUBY=true
PATH=/opt/sensu/embedded/:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/etc/sensu/plugins
. /etc/sensu/stackrc
[root@controller-0 checks]# . /etc/default/sensu
[root@controller-0 checks]# which ruby
/opt/sensu/embedded/ruby
[root@controller-0 checks]# /etc/sensu/plugins/metrics-net.rb --scheme stats.bbc.ci-full-centos.controller-0.novalocal.network
stats.bbc.ci-full-centos.controller-0.novalocal.network.eth0.tx_packets 151495566 1484287692
stats.bbc.ci-full-centos.controller-0.novalocal.network.eth0.rx_packets 172666363 1484287692
stats.bbc.ci-full-centos.controller-0.novalocal.network.eth0.tx_bytes 28989186646 1484287692
stats.bbc.ci-full-centos.controller-0.novalocal.network.eth0.rx_bytes 23230760262 1484287692
stats.bbc.ci-full-centos.controller-0.novalocal.network.eth0.tx_errors 0 1484287692
stats.bbc.ci-full-centos.controller-0.novalocal.network.eth0.rx_errors 0 1484287692
stats.bbc.ci-full-centos.controller-0.novalocal.network.eth0.tx_dropped 0 1484287692
stats.bbc.ci-full-centos.controller-0.novalocal.network.eth0.rx_dropped 0 1484287692
```
I removed '/opt/sensu/embedded/' change in this PR, and add one more PR in usula.
https://github.com/blueboxgroup/ursula/pull/2382
please review and merge both of them

Xiaohua-Shen committed:
commit

1. no ifquery command in centos/rhel
2. clean the output of check-cinder-service (RTC defect  294012)

Xiaohua-Shen commented:
one more commit for:
1. no ifquery command in centos/rhel
2. clean the output of check-cinder-service (RTC defect  294012)
